m2cgen/interpreters/mixins.py

```diff
@@ -195,3 +195,46 @@ def _get_subroutine_name(self):
 
     def create_code_generator(self):
         raise NotImplementedError
+
+
+class InfixPowExprMixin(BaseToCodeInterpreter):
+    """
+    This mixin is used for languages that provide the exponentiation operation in a form
+    of an infix operator. Such languages inlcude Haskell, F#, R and others.
+
+    The operator used by default is "**", but it can be overriden by setting an appropriate
+    value in `pow_operator`.
+    """
+
+    pow_operator = "**"
+
+    infix_expressions = [*BaseToCodeInterpreter.infix_expressions, ast.PowExpr]
+
+    def interpret_pow_expr(self, expr, left_precedence=None,
+                           right_precedence=None, **kwargs):
+        base_result = self._do_interpret(
+            expr.base_expr, left_precedence=expr.precedence, **kwargs)
+        exp_result = self._do_interpret(
+            expr.exp_expr, right_precedence=expr.precedence, **kwargs)
+        return self._cg.infix_expression(
+            left=base_result, right=exp_result, op=self.pow_operator,
+            wrap=self._wrap_infix_expr(expr, left_precedence, right_precedence))
+
+
+class FunctionPowExprMixin(BaseToCodeInterpreter):
```

Comment:
Maybe make one of these two mixins as a default behavior? This will allow contributors to not get familiar with mixins mechanism in a half (roughly, of course) of all cases.

Reply:
So I actually did this intentionally :) I want the developer to make a conscious choice and pick one or the other. 

> Maybe make one of these two mixins as a default behavior? 

Well, I don't think there is such thing as a "default" behavior in this case, since even with the pow expression as a function a potential contributor needs to provide a value for the `pow_function_name` which doesn't have a default. So I'd rather prefer this choice to be explicit.

Reply:
OK, fine with me.